src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class type) {
+        if (type.equals(Bulldozer.class)) {
+            return new Bulldozer().get();
```

Comment:
```suggestion
            return new ArrayList(bulldozerProducer.get());
```
same for other cases